### DISCUSSION TITLE
plc.list_blocks()  CPU : Function not available (S7-PLCSIM Advanced v4.0 S7-1500)

### DISCUSSION THREAD

**Discussion by Motorov** [gijzelaerr/python-snap7#390] [Q&A] [answered]
ord= plc.list_blocks()
print(ord)

Traceback (most recent call last):
  File "c:\python_code\test.py", line 20, in <module>
    ord= plc.list_blocks()
  File "C:\Python\lib\site-packages\snap7\client.py", line 468, in list_blocks
    check_error(result, context="client")
  File "C:\Python\lib\site-packages\snap7\common.py", line 89, in check_error 
    raise RuntimeError(error)
RuntimeError: b'CPU : Function not available'

It looks like not all features are supported for the 1500 controller?

**Comment by swamper123** [comment]
Yes, not everything is supported by the snap7 library (see [here](http://snap7.sourceforge.net/snap7_client.html#target_compatibility)) for 1200/1500 (they need a special config as well, like mentioned in the link).

Not sure if list_blocks() is one of them tbh.

**Reply by swamper123** [reply, marked as answer]
Your 1500 is configured like the docs mentioned for 1200/1500 (see the link)?
If yes, then it is not possible atm. and this is the answer from PLC side.

Since python-snap7 is only a wrapper for snap7, a workaround is difficult for some stuff and I guess for this as well. :/
But PRs are always welcomed for new feature :)

**Comment by gijzelaerr** [comment]
Closing as answered. This is a known S7-1200/1500 compatibility limitation. Not all functions are supported on these PLCs.
